src/scripts/main.js

```diff
@@ -1,3 +1,34 @@
 'use strict';
 
 // write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const headers = document.querySelectorAll('th');
+  const tableBody = document.querySelector('tbody');
+
+  const rows = Array.from(tableBody.querySelectorAll('tr'));
+
+  headers.forEach((header, index) => {
+    header.addEventListener('click', () => {
+      const sortedRows = [...rows].sort((a, b) => {
+        const aText = a.cells[index].textContent.trim();
+        const bText = b.cells[index].textContent.trim();
```

Comment:
You are calling `trim()` on the text content of each cell every time you compare two rows. This could be inefficient if you have a large table because it will be called multiple times for the same cell. You might consider trimming the text content once before sorting.